tests/test_ecu_variant_matching.py

```diff
@@ -174,15 +238,31 @@ def ecu_variant_3(
     ecu_variant_pattern1: EcuVariantPattern,
     ecu_variant_pattern3: EcuVariantPattern,
 ) -> DiagLayer:
-    return DiagLayer(
+    result = DiagLayer(
         variant_type=DIAG_LAYER_TYPE.ECU_VARIANT,
         odx_id=OdxLinkId(local_id="ecu_variant3", doc_fragments=doc_frags),
         short_name="ecu_variant3",
+        long_name=None,
+        description=None,
+        functional_classes=[],
+        additional_audiences=[],
+        parent_refs=[],
+        import_refs=[],
+        communication_parameters=[],
+        diag_comm_refs=[],
+        diag_data_dictionary_spec=None,
         services=[ident_service, supplier_service],
+        requests=[],
+        positive_responses=[],
+        negative_responses=[],
+        single_ecu_jobs=[],
+        states=[],
```

Comment:
Testers do not like this change ;-)

But seriously, I think it is acceptable since the constructor of DiagLayer is usually not part of the public interface.